ui/app/src/views/ViewDashboard.tsx

```
@@ -11,32 +11,78 @@
+  PersesChartsTheme,
+} from '@perses-dev/components';
+import { useMemo } from 'react';
+import { PluginRegistry } from '@perses-dev/plugin-system';
+import { useBundledPlugins } from '../model/bundled-plugins';
+import { useDashboard } from '../model/dashboard-client';
 import { useDatasourceApi } from '../model/datasource-api';
-import { useSampleData } from '../utils/temp-sample-data';
 
-const DEFAULT_DASHBOARD_ID = 'node-exporter-full';
+// app specific echarts option overrides, empty since perses uses default
+// https://apache.github.io/echarts-handbook/en/concepts/style/#theme
+const ECHARTS_THEME_OVERRIDES = {};
+
+interface ViewSingleDashboardProperty {
+  dashboardName: string;
+  projectName: string;
+}
+
+function ViewSingleDashboard(props: ViewSingleDashboardProperty) {
+  const { dashboardName, projectName } = props;
+  // TODO: Loading indicator
+  const { data, isLoading } = useDashboard(projectName, dashboardName);
+  const datasourceApi = useDatasourceApi();
+
+  if (!isLoading && data !== undefined) {
+    return <DashboardView dashboardResource={data} datasourceApi={datasourceApi} />;
+  }
+  return null;
+}
 
 /**
  * The View for viewing a Dashboard.
  */
 function ViewDashboard() {
-  const [query] = useQueryParams({ dashboard: StringParam });
-  const { dashboard } = query;
-
-  // TODO: setup project routing
-  const sampleDashboard = useSampleData<DashboardResource>(dashboard || DEFAULT_DASHBOARD_ID);
+  const { getInstalledPlugins, importPluginModule } = useBundledPlugins();
+  const muiTheme = useTheme();
+  const chartsTheme: PersesChartsTheme = useMemo(() => {
+    return generateChartsTheme('perses', muiTheme, ECHARTS_THEME_OVERRIDES);
+  }, [muiTheme]);
 
-  const datasourceApi = useDatasourceApi();
-
-  // TODO: Loading indicator
-  if (sampleDashboard === undefined) {
+  const projectName = useParams().projectID;
+  const dashboardName = useParams().dashboardID;
+  if (projectName === undefined || dashboardName === undefined) {
```

Assuming we can just some other view as the "home page" (i.e. when we don't have a project or dashboard), we should probably just throw an error in this case.

---

thanks for the tips. It's much cleaner like that indeed !